### PR TITLE
Feature/grpc faas

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: proto/faas/v1/faas.proto [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+// protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.faas.v1";
+option java_multiple_files = true;
+option java_outer_classname = "NitricFaas";
+option php_namespace = "Nitric\\Proto\\Faas\\V1";
+option csharp_namespace = "Nitric.Proto.Faas.v1";
+
+// Service for management of event topics
```
Comment (jyecusch) [inline proto/faas/v1/faas.proto:12]: need to update the comments to match this service and its messages, a few copy/paste hangovers are left behind.

File: proto/faas/v1/faas.proto [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+// Placeholder message
+message InitRequest {}
+
+// Placeholder message
+message InitResponse {}
+
+
+// The membrane has a trigger for this function to handle
```
Comment (jyecusch) [inline proto/faas/v1/faas.proto:57]: Membrane has a trigger for the *client* to handle

File: proto/faas/v1/faas.proto [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+// The worker has successfully processed a trigger
+message TriggerResponse {
+  // The data returned in the response
+  bytes data = 1;
+
+  // The context of the request response
+  // Typically this will be one to one with the Trigger Context
+  // i.e. if you recieve http context you may return http context
```
Comment (jyecusch) [inline proto/faas/v1/faas.proto:104]: *receive*

File: proto/faas/v1/faas.proto [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+message HttpResponseContext {
+  // The request headers...
+  map<string, string> headers = 1;
+  // The HTTP status of the request
+  int32 status = 2;
+}
+
+// Specific event response message
+// We do not accept responses for events
+// only whether or not they were successfully responded to
```
Comment (jyecusch) [inline proto/faas/v1/faas.proto:126]: successfully *processed*

**Review by jyecusch — CHANGES_REQUESTED** [review]

File: proto/faas/v1/faas.proto (hunk 1 above)
Comment (tjholm) [reply proto/faas/v1/faas.proto:12]: fixed.

File: proto/faas/v1/faas.proto (hunk 2 above)
Comment (tjholm) [reply proto/faas/v1/faas.proto:57]: fixed.

File: proto/faas/v1/faas.proto (hunk 4 above)
Comment (tjholm) [reply proto/faas/v1/faas.proto:126]: fixed.

**Review by jyecusch — APPROVED** [review]

**Pull request merged**